src/model/wallet.js

```diff
@@ -368,15 +385,44 @@ export const createWallet = async (seed = null, color = null, name = null) => {
         const hasTxHistory = await ethereumUtils.hasPreviousTransactions(
           nextWallet.address
         );
+
+        let discoveredAccount = null;
+        let discoveredWalletId = null;
+        find(allWallets, someWallet => {
+          const existingAccount = find(
+            someWallet.addresses,
+            account =>
+              toChecksumAddress(account.address) ===
+              toChecksumAddress(nextWallet.address)
```

Comment:
We should add `&& account.visible` to prevent replacing wallets because it matched against a deleted one (not visible)